Compiling in MSVS 2019

simplyWiri opened the pull request:
Fixed majority of the compilation errors / linking errors;
- Source is now dependant on net 4.7.2
- Added additional unity components:
 UnityEngine.
   - CoreModule, IMGUIModule, InputLegacyModule, InputModule, TextRenderingModule
- Re-Added Harmony 
Harmony is now linked as a mod dependency, in the same fashion as the HugsLib dependency. I have not touched the XML for this dependancy, nor folder structure.

Do you want to open a WIP-1.1 branch for this, as appose to merging into what being used as the master branch?

(I'm a student mid sem, by no means a C# developer, nor very experienced with Rimworld defs/names. I don't have tons of time spare, but i'll happily help where I can/when I have time)

simplyWiri committed:
Compilation

Compiling in MSVS 2019

CTRL SHIFT F "@@@@" for notes I leave.

simplyWiri committed:
Following Standard

Updated to use already existing project standards.

Not sure what HugsLibChecker did?

simplyWiri committed:
Repair DLL location

simplyWiri committed:
Cleanup

ghost commented:
<img src="https://www.deepcode.ai/icons/green_check.svg" width= "50px" align= "left"/> Congratulations :beers:. DeepCode [analyzed](https://www.deepcode.ai/app/gh/OrionFive/Hospitality/66ce6c560a02bdff3ba0b6fb25ab93d052c19d29/OrionFive/Hospitality/5d233d36778ded9ea6e729a89966ebad11cc1804/_/%2F/code/?utm_source=gh_review) your code in 0.105 seconds and we found no issues. Enjoy a moment of no bugs :sunny:.

:speech_balloon: This comment has been generated by the [**DeepCode bot**](https://github.com/apps/deepcode-ci-bot/), installed by the owner of the repository. The DeepCode bot **protects** your repository by detecting and commenting on security vulnerabilities or other critical issues.

---
:relaxed: If you want to provide feedback on our bot, here is how to [contact us](https://www.deepcode.ai/feedback?select=4&utm_source=gh_review).

OrionFive reviewed (commented):
Thanks, it's the easy part, but it will save me some time nonetheless!
I'll go through it again in more detail when I have time. I can tell the ownersForReading thing will not work like that.

OrionFive commented on Source/Source/CompGuest.cs:
```diff
@@ -93,7 +94,7 @@ public override void PostExposeData()
         internal void ClearOwnership()
         {
-            bed?.owners.Remove(Pawn);
+            bed?.OwnersForReading.Remove(Pawn);
```
Won't work

OrionFive commented on Source/Source/Building_GuestBed.cs:
```diff
@@ -53,10 +53,10 @@ public override void Draw()
             {
                 owner.ownership.UnclaimBed();
-                owners.Clear();
+                OwnersForReading.Clear();
```
Won't work

OrionFive commented on Source/Source/CompGuest.cs:
```diff
@@ -132,11 +133,13 @@ public void ClaimBed(Building_GuestBed newBed)
-            newBed.owners.Add(Pawn);
+            // @@@@Attention
+            // not sure how this will work as its probably intended as a readonly
+            newBed.OwnersForReading.Add(Pawn);
```
Won't work

OrionFive commented on .vs/ProjectSettings.json:
```diff
@@ -0,0 +1,3 @@
+{
+  "CurrentProjectSetting": null
```
Please don't include files from .vs

OrionFive reviewed (changes requested):
I've marked some obvious items. Sorry for the brevity, I'm doing it from my phone at work.

simplyWiri committed:
Changes

+ Using field
CompAssignableToPawn instead of readable owners when data is changed. Using functions provided by the class.
- Removed vs files, not sure when they got past the gitignore

simplyWiri committed:
Fixed hugslib dep

Using 1.1 version of hugslib dll

simplyWiri committed:
No Bizarre Compilation Errors

Got rid of all compilation errors regarding xml setup
- Hugslib + Harmony are both 'required' (Hugslib requires Harmony, so you may find it redundant)
- The mod will now also be 'auto sorted' below both of the required mods.

I've tagged the harmony patch which is failing upon loading the mod
